agenta-cli/agenta/sdk/config_manager.py

```diff
@@ -0,0 +1,205 @@
+        elif "environment" in context and context["environment"]:
+            return ConfigManager.get_from_backend(
+                schema, environment=context["environment"]
+            )
+        elif "variant" in context and context["variant"]:
+            return ConfigManager.get_from_backend(schema, variant=context["variant"])
+        else:
+            raise ValueError("Either config, environment or variant must be provided")
+
+    @staticmethod
+    def get_from_backend(
+        schema: Type[T],
+        environment: Optional[str] = None,
+        version: Optional[str] = None,
+        variant: Optional[str] = None,
+    ) -> T:
+        """
+        Pulls the parameters for the app variant from the server and returns a config object.
+
+        This method retrieves the configuration from the backend server based on the provided
+        environment or variant. It then validates and returns the configuration as an instance
+        of the specified schema.
+
+        Args:
+            schema (Type[T]): A Pydantic model class that defines the structure of the configuration.
+            environment (Optional[str]): The environment name to fetch the configuration for.
+                                         Must be one of "development", "production", or "staging".
+            version (Optional[str]): Currently not implemented. Will raise NotImplementedError if provided.
+            variant (Optional[str]): The variant name to fetch the configuration for.
+
+        Returns:
+            T: An instance of the specified schema populated with the configuration data.
+
+        Raises:
+            ValueError: If neither environment nor variant is provided.
+            NotImplementedError: If a specific version is requested (not yet implemented).
+            ValidationError: If the retrieved configuration data doesn't match the schema.
+            Exception: For any other errors during the process (e.g., API communication issues).
+
+        Note:
+            Either environment or variant must be provided, but not both.
+        """
+        if not ConfigManager.client:
+            try:
+                ConfigManager.client = AgentaApi(
+                    base_url=singleton.host + "/api",
+                    api_key=singleton.api_key if singleton.api_key else "",
+                )
+            except Exception as ex:
+                logger.error(
+                    "Failed to initialize Agenta client with error: %s", str(ex)
+                )
+                raise
+        if not environment and not variant:
+            raise ValueError("Either environment or variant must be provided")
+        try:
+            if environment:
+                if version:
+                    raise NotImplementedError(
+                        "Getting config for a specific version is not implemented yet."
+                    )
+                else:
+                    assert environment in [
```

Comment:
We might want to define this list of environment in a const at the top, like `AVAILABLE_ENVIRONMENTS`